### PR TITLE
Don't drop RUM unsampled transactions

### LINKED ISSUE #6260: [RUM] Keep unsampled transactions for the rum intake
APM server is introducing a breaking change in 8.0 - https://github.com/elastic/apm-server/issues/5997

As the User Experience Dashboard and Exploratory UI is built based on transaction events and not taking advantage of the transaction metrics documents at the moment, We(synthetics team) still need a way to keep the un-sampled transaction for all RUM events and also probably for agents(Ex- mobile agents iOS) by default  that are part of the UX and Exploratory view app. 

Discussion from [slack thread](https://elastic.slack.com/archives/C877NKGBY/p1632829520249700)
```
Vignesh Shanmugam  1 day ago
As Andrew is on vacation. I am delegating this question to @silvia Is there a way to exclude RUM events from getting dropped as part of the metrics initiative?

Silvia Mitter  1 day ago
not directly; should it be configurable, or never be applied to RUM events?

Silvia Mitter  1 day ago
Dropping samples is done in a dedicated processor; excluding all events from a certain agent seems doable. @v can you open an issue with the description whether this should always be excluded or be configurable?

Silvia Mitter  1 day ago
@bryce.buchanan is dropping unsampled events also a concern for iOS?

Vignesh Shanmugam  1 day ago
Sure let me bring it up on our refinement call and will do post that. Thanks for double checking @silvia

Bryce Buchanan:flag-us:  1 day ago
we don’t have any customized UI for the iOS agent at this point, so I think whatever implications this has for the standard apm agents apply to the iOS agent. I’m not familiar with what the ‘UX app’ is, however.

```

**Note: Mobile agents is just an example, this issue is more focussed towards RUM User Experience app at the moment**

Open questions (on the UI side)
- Would we still be able to link between APM UI and UX app in terms of transactions and errors

@paulb-elastic @drewpost

### PR DESCRIPTION
**Pull request by axw**
## Motivation/summary

The User Experience app in Kibana currently relies on transaction events only, and not transaction metrics. Store all RUM transaction events so the UX app can calculate statistics correctly.

## Checklist

- [x] Update [CHANGELOG.asciidoc](https://github.com/elastic/apm-server/blob/master/CHANGELOG.asciidoc)
~- [ ] Update [package changelog.yml](https://github.com/elastic/apm-server/blob/master/apmpackage/apm/changelog.yml) (only if changes to `apmpackage` have been made)~
~- [ ] Documentation has been updated~

## How to test these changes

1. Run apm-server, send unsampled RUM transactions and unsampled backend transactions
2. Check that only the RUM transactions are stored

## Related issues

Closes #6260

### CODE REVIEW

**Comment by axw** [comment]
This is broken, moving back into draft for now.

File: beater/beater.go [hunk 1]
```diff
@@ -535,7 +534,7 @@ func (s *serverRunner) run(listener net.Listener) error {
 	batchProcessor = append(batchProcessor,
 		// The server always discards unsampled transactions. It is important that this
 		// is done just before calling the publisher to avoid affecting aggregations.
-		sampling.NewDiscardUnsampledBatchProcessor(),
+		modelprocessor.NewDropUnsampled(false /* don't drop RUM unsampled transactions*/),
```
[the remark this answers is not in the document]
Comment (axw) [reply beater/beater.go:541]: Done, thanks

**Review by marclop — APPROVED** [review]

File: changelogs/head.asciidoc [hunk 2]
```diff
@@ -23,7 +23,7 @@ https://github.com/elastic/apm-server/compare/7.15\...master[View commits]
 - Removed unsupported libbeat `processors` configuration {pull}6474[6474]
 - Removed `apm-server.aggregation.transactions.enabled` configuration option {pull}6495[6495]
 - Removed `apm-server.aggregation.service_destinations.enabled` configuration option {pull}6503[6503]
-- Removed `apm-server.sampling.keep_unsampled` configuration option {pull}6514[6514]
+- Removed `apm-server.sampling.keep_unsampled` configuration option; non-RUM unsampled transactions are always dropped {pull}6514[6514]
```
Comment (simitt) [inline changelogs/head.asciidoc:26]: Please add this PR number to the changelog entry

**Review by simitt — APPROVED** [review]

**Comment by marclop** [comment]
Verified with 9bb5e4cdd:

Ingested transactions that are both sampled and upsampled, the "normal" unsampled transactions are dropped:

```
$ curl -H "Content-type: application/x-ndjson" --data-binary @testdata/intake-v2/transactions_spans.ndjson http://localhost:8200/intake/v2/events
$ grep 'sampled":false' testdata/intake-v2/transactions_spans.ndjson
{"transaction":{"id":"85925e55b43f4341","name":"GET /api/types","type":"request","duration":13.980558,"result":"200","timestamp":1496170422000000,"sampled":false,"trace_id":"85925e55b43f4341aaaaaaaaaaaaaaaa","span_count":{"started":0},"context":{"service":null}}}
```

```javascript
GET traces-apm*/_search
{
  "query": {
    "match": {
      "transaction.id": "85925e55b43f4341"
    }
  }
}

// ...returns no results.
```

Ingested upsampled rum transactions:

```json
{"metadata":{"service":{"name":"apm-agent-js","version":"1.0.0","agent":{"name":"rum-js","version":"0.0.0"}},"network":{"connection":{"type":"5G"}}}}
{"transaction":{"sampled":false,"id":"611f4fa950f04632","type":"page-load","duration":643,"context":{"page":{"referer":"http://localhost:8000/test/e2e/","url":"http://localhost:8000/test/e2e/general-usecase/"}},"trace_id":"611f4fa950f04631aaaaaaaaaaaaaaaa","span_count":{"started":1},"experience":{"cls":1,"fid":2.0,"tbt":3.4,"ignored":5,"also":"ignored"}}}
{"span":{"name":"transaction","type":"transaction","start":0,"duration":643,"stacktrace":[{"abs_path":"http://localhost:8000/test/e2e/general-usecase/bundle.js.map","filename":"test/e2e/general-usecase/bundle.js.map","function":"<anonymous>","lineno":1,"colno":18},{"abs_path":"http://localhost:8000/test/e2e/general-usecase/bundle.js.map","filename":"~/test/e2e/general-usecase/bundle.js.map","function":"<anonymous>","lineno":1,"colno":18}],"context":{"http":{"url":"http://localhost:8000/test/e2e/general-usecase/span"}},"transaction_id":"611f4fa950f04631","parent_id":"611f4fa950f04631","trace_id":"611f4fa950f04631aaaaaaaaaaaaaaaa","id":"aaaaaaaaaaaaaaaa"}}
```

```javascript
GET traces-apm.rum*/_search
{
  "query": {
    "match": {
      "transaction.id": "611f4fa950f04632"
    }
  }
}
```

```json
{
  "took" : 127,
  "timed_out" : false,
  "_shards" : {
    "total" : 1,
    "successful" : 1,
    "skipped" : 0,
    "failed" : 0
  },
  "hits" : {
    "total" : {
      "value" : 1,
      "relation" : "eq"
    },
    "max_score" : 1.7917595,
    "hits" : [
      {
        "_index" : ".ds-traces-apm.rum-default-2021.12.21-000001",
        "_id" : "rDP6230ByDpadNqDuU9C",
        "_score" : 1.7917595,
        "_source" : {
          "agent" : {
            "name" : "rum-js",
            "version" : "0.0.0"
          },
          "data_stream.namespace" : "default",
          "source" : {
            "port" : 53042,
            "ip" : "127.0.0.1"
          },
          "data_stream.type" : "traces",
          "processor" : {
            "name" : "transaction",
            "event" : "transaction"
          },
          "url" : {
            "path" : "/test/e2e/general-usecase/",
            "original" : "http://localhost:8000/test/e2e/general-usecase/",
            "scheme" : "http",
            "port" : 8000,
            "domain" : "localhost",
            "full" : "http://localhost:8000/test/e2e/general-usecase/"
          },
          "network" : {
            "connection" : {
              "type" : "5G"
            }
          },
          "observer" : {
            "hostname" : "Marcs-MacBook-Pro-2.local",
            "id" : "bf2a6d26-c5ed-4b41-a507-feab5b75c095",
            "type" : "apm-server",
            "ephemeral_id" : "fdeb69ec-dad9-4791-b23f-022c3948ec37",
            "version" : "8.0.0",
            "version_major" : 8
          },
          "trace" : {
            "id" : "611f4fa950f04631aaaaaaaaaaaaaaaa"
          },
          "@timestamp" : "2021-12-21T15:53:00.481+08:00",
          "ecs" : {
            "version" : "1.12.0"
          },
          "service" : {
            "name" : "apm-agent-js",
            "version" : "1.0.0"
          },
          "data_stream.dataset" : "apm.rum",
          "http" : {
            "request" : {
              "referrer" : "http://localhost:8000/test/e2e/"
            }
          },
          "client" : {
            "ip" : "127.0.0.1"
          },
          "event" : {
            "agent_id_status" : "missing",
            "ingested" : "2021-12-21T07:53:01Z",
            "outcome" : "unknown"
          },
          "user_agent" : {
            "original" : "curl/7.64.1",
            "name" : "curl",
            "device" : {
              "name" : "Other"
            },
            "version" : "7.64.1"
          },
          "transaction" : {
            "duration" : {
              "us" : 643000
            },
            "id" : "611f4fa950f04632",
            "span_count" : {
              "started" : 1
            },
            "type" : "page-load",
            "experience" : {
              "fid" : 2,
              "tbt" : 3.4,
              "cls" : 1
            }
          },
          "timestamp" : {
            "us" : 1640073180481075
          }
        }
      }
    ]
  }
}
```

**Pull request merged**